analysis/quantifier-comparisons/08-genes-detected-comparison.Rmd

```diff
@@ -0,0 +1,405 @@
+  dplyr::pull(gene_id) %>%
+  unique()
+cellranger_unique <- setdiff(cellranger_genes, all_cellranger_overlap)
+
+cr_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cr_like_genes & alevin_resolution %in% c("cr-like-em", "not_alevin")) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cr_unique <- setdiff(cr_like_genes, cr_overlap)
+
+cr_like_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cr_em_genes & alevin_resolution %in% c("cr-like", "not_alevin")) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cr_like_unique <- setdiff(cr_em_genes, cr_like_overlap)
+```
+
+
+```{r}
+# visualize overlap between gene sets based on tool
+rowdata_filtered[which(rowdata_filtered$alevin_resolution == "not_alevin"), "alevin_resolution"] <- "cellranger"
+
+gene_detect_df <- rowdata_filtered %>%
+  group_by(gene_id) %>%
+  dplyr::summarise(tools_detected = list(unique(alevin_resolution)))
+
+gene_detect_df %>%
+  dplyr::distinct(gene_id, .keep_all = TRUE) %>%
+  ggplot(aes(x = tools_detected)) +
+  geom_bar() +
+  scale_x_upset(n_intersections = 6)
+```
+
+
+When looking at shared cells only and genes that are expressed and detected > 0, we see that alevin-fry detects more genes than cellranger in both methods. 
+Almost every gene that is found in cellranger is also being found in Alevin-fry so we are probably not losing much.
+We also see here that there are more in `cr-like-em` than in `cr-like`. 
+
+Perhaps the genes that are different are lowly expressed genes and it wouldn't matter anyway. 
+
+If we now look at the gene expression for those genes that are unique to each tool, how do they compare to the gene expression for those genes that are shared?
+
+
+```{r}
+# calculate mean expression of log transformed counts matrix 
+log_row_means <- function(sce){
+   rowMeans(log1p(counts(sce))) %>%
+    as.data.frame() %>%
+    tibble::rownames_to_column("gene_id") %>%
+    dplyr::rename("mean" = ".")
+}
+
+# get list of row mean data frames 
+mean_exp_list <- sces_filtered %>%
+  purrr::map(
+    ~ purrr::map(.x, log_row_means)
+  )
+
+# join together into larger data frame with quant_dir and alevin_resolution as unique ID's
+mean_exp_df <- mean_exp_list %>%
+  purrr::map_df(
+    ~ purrr::map_df(.x, rbind, .id = "quant_dir"), 
+    .id = "alevin_resolution"
+  ) %>%
+  dplyr::left_join(quant_info)
+
+# join back with quant_info to get sample information 
+```
+
+
+```{r}
+# add a column to classify as shared or unique to that tool
+mean_exp_df <- mean_exp_df %>%
+  dplyr::mutate(gene_status = ifelse(gene_id %in% c(cellranger_cr_intersect, cellranger_cr_em_intersect, cr_cr_em_intersect), "shared", "unique")) 
+```
+
+
+```{r fig.height = 5, fig.width=10}
+# make a plot to look at the mean gene expression of shared vs. unique genes for each tool
+ggplot(mean_exp_df, aes(x = alevin_resolution, y = mean, fill = gene_status)) + 
+  geom_boxplot() + 
+  facet_grid(~sample) + 
+  coord_cartesian(ylim = c(0,2))
```

Comment:
Now that we are looking at log expression, we probably want to zoom in more.. maybe to 1?